libs/platform/experience/entity-text/entity-text.component.ts

```diff
@@ -1,28 +1,87 @@
-import { EntityService } from '@spryker-oryx/core';
+import { ContextService, EntityService } from '@spryker-oryx/core';
 import { resolve } from '@spryker-oryx/di';
 import { ContentMixin } from '@spryker-oryx/experience';
-import { computed, hydrate, Signal } from '@spryker-oryx/utilities';
+import { LinkService } from '@spryker-oryx/router';
+import { TextMixin } from '@spryker-oryx/ui/text';
+import { Signal, computed, hydrate } from '@spryker-oryx/utilities';
 import { LitElement, TemplateResult } from 'lit';
+import { DirectiveResult } from 'lit/directive';
+import { UnsafeHTMLDirective } from 'lit/directives/unsafe-html';
 import { html } from 'lit/static-html.js';
 import { EntityTextOptions } from './entity-text.model';
 
 @hydrate()
-export class EntityTextComponent extends ContentMixin<EntityTextOptions>(
-  LitElement
+export class EntityTextComponent extends TextMixin(
+  ContentMixin<EntityTextOptions>(LitElement)
 ) {
   protected entityService = resolve(EntityService);
+  protected linkService = resolve(LinkService);
+  protected contextService = resolve(ContextService);
 
-  protected $entityField: Signal<string | undefined> = computed<
-    string | undefined
-  >(() => {
-    return this.entityService.getField<string>({
-      element: this,
-      type: this.$options().entity,
-      field: this.$options().field,
+  protected $data: Signal<string | undefined> = computed<string | undefined>(
+    () => {
+      const { entity: type, field } = this.$options();
+      return this.entityService.getField({ element: this, type, field });
+    }
+  );
+
+  protected $text = computed(() => {
+    const text = this.$data();
+    if (!text) return;
+    return this.convertText(text);
+  });
+
+  protected $link = computed(() => {
+    const { entity, field } = this.$options();
+    if (!entity || !field) return;
+
+    const qualifier = this.contextService.get(this, field);
+    return this.linkService.get({
+      type: entity,
+      qualifier,
     });
   });
 
   protected override render(): TemplateResult | void {
-    return html`${this.$entityField()}`;
+    const content = this.renderContent();
+    if (!content) return;
+    const { tag } = this.$options();
+    if (tag) {
+      return html`<oryx-heading .tag=${tag}>${content}</oryx-heading>`;
+    } else {
+      return content;
+    }
+  }
+
+  protected renderContent(): TemplateResult | void {
+    const text = this.$text();
+    if (!text) return;
+    // const { link } = this.$options();
+    // const content = html`${this.renderPrefix()}${link
+    //   ? this.renderLink()
+    //   : this.$text()} `;
+
+    return html`${this.renderPrefix()}${this.$text()}`;
+  }
+
+  protected renderLink():
+    | TemplateResult
+    | DirectiveResult<typeof UnsafeHTMLDirective>
```

Comment:
`DirectiveResult<typeof UnsafeHTMLDirective>`, what condition returns this ?

Reply:
`this.$text();`